test/state.js

```diff
@@ -42,3 +45,33 @@ test('cursor set', t => {
   cursorState.set('c.e', 2);
   t.deepEqual(state.get(), { a: { b: { c: { d: 1, e: 2 } } } });
 });
+
+test('event emit', t => {
+  t.plan(4);
+  const state = new State();
+  process.nextTick(() => {
+    state.set('', 1);
+    state.set('a', 1);
+    state.set('a.b', 1);
+    state.set('c.d', 1);
+  });
+  return Observable
+    .from(state.listen('change'))
+    .map(() => t.pass())
+    .timeoutWith(OBSERVABLE_TIMEOUT, Observable.empty());
+});
+
+test('event emit with cursor', t => {
+  t.plan(3);
+  const state = new State();
+  process.nextTick(() => {
+    state.set('', 1);
+    state.set('a', 1);
+    state.set('a.b', 1);
```

Comment:
yes and i fixed the test.
https://github.com/nofluxjs/state/pull/3/commits/50b840986794fd7a08f75090262837493e04fab3#diff-f13dfcc101d7d9daad638042913a3418R73